geometry/optimization/graph_of_convex_sets.cc

```diff
@@ -258,12 +263,164 @@ std::string GraphOfConvexSets::GetGraphvizString(
+  std::map<VertexId, std::vector<int>> outgoing_edges;
+  std::set<EdgeId> unusable_edges;
+
+  int edge_count = 0;
+  for (const auto& [edge_id, e] : edges_) {
+    outgoing_edges[e->u().id()].push_back(edge_count);
+    incoming_edges[e->v().id()].push_back(edge_count);
+
+    // Turn off edges into source or out of target
+    if (e->v().id() == source_id || e->u().id() == target_id) {
+      unusable_edges.insert(edge_id);
+    }
+    edge_count++;
+  }
+
+  int nE = edges_.size();
+
+  // Given an edge (u,v) check if a path from source to u and another from v to
+  // target exist without sharing edges.
+  MathematicalProgram prog;
+
+  // Flow for each edge is between 0 and 1 for both paths.
+  VectorXDecisionVariable f = prog.NewContinuousVariables(nE, "flow_su");
+  Binding<solvers::BoundingBoxConstraint> f_limits =
+      prog.AddBoundingBoxConstraint(0, 1, f);
+  VectorXDecisionVariable g = prog.NewContinuousVariables(nE, "flow_vt");
+  Binding<solvers::BoundingBoxConstraint> g_limits =
+      prog.AddBoundingBoxConstraint(0, 1, g);
+
+  std::map<VertexId, Binding<LinearEqualityConstraint>> conservation_f;
+  std::map<VertexId, Binding<LinearEqualityConstraint>> conservation_g;
+  std::map<VertexId, Binding<LinearConstraint>> degree;
+  for (const auto& [vertex_id, v] : vertices_) {
+    std::vector<int> Ev_in = incoming_edges[vertex_id];
+    std::vector<int> Ev_out = outgoing_edges[vertex_id];
+    std::vector<int> Ev = Ev_in;
+    Ev.insert(Ev.end(), Ev_out.begin(), Ev_out.end());
+
+    RowVectorXd A_flow(Ev.size());
+    A_flow << RowVectorXd::Ones(Ev_in.size()),
+        -1 * RowVectorXd::Ones(Ev_out.size());
+    VectorXDecisionVariable fv(Ev.size());
+    VectorXDecisionVariable gv(Ev.size());
+    for (size_t ii = 0; ii < Ev.size(); ++ii) {
+      fv(ii) = f(Ev[ii]);
+      gv(ii) = g(Ev[ii]);
+    }
+
+    // Conservation of flow for f: ∑ f_in - ∑ f_out = -δ(is_source).
+    if (vertex_id == source_id) {
+      conservation_f.insert(
+          {vertex_id, prog.AddLinearEqualityConstraint(A_flow, -1, fv)});
+    } else {
+      conservation_f.insert(
+          {vertex_id, prog.AddLinearEqualityConstraint(A_flow, 0, fv)});
+    }
+
+    // Conservation of flow for g: ∑ g_in - ∑ g_out = δ(is_target).
+    if (vertex_id == target_id) {
+      conservation_g.insert(
+          {vertex_id, prog.AddLinearEqualityConstraint(A_flow, 1, gv)});
+    } else {
+      conservation_g.insert(
+          {vertex_id, prog.AddLinearEqualityConstraint(A_flow, 0, gv)});
+    }
+
+    // Degree constraints (redundant if indegree of w is 0):
+    // 0 <= ∑ f_in + ∑ g_in <= 1
+    RowVectorXd A_degree = RowVectorXd::Ones(2 * Ev_in.size());
+    VectorXDecisionVariable fgin(2 * Ev_in.size());
+    for (size_t ii = 0; ii < Ev_in.size(); ++ii) {
+      fgin(ii) = f(Ev_in[ii]);
+      fgin(Ev_in.size() + ii) = g(Ev_in[ii]);
+    }
+    degree.insert({vertex_id, prog.AddLinearConstraint(A_degree, 0, 1, fgin)});
```

Comment:
I would think that adding an explicit lower bound of zero here should not help, rather slow down? (This lower bound is implied by the non negativity of the flows, right?) I can imagine Gurobi and Mosek being faster if you give an explicit redundant bound on a variable, but I think that adding redundant constraints of linear combinations of decision variables should not help.